packages/web/src/pages/oauth-login-page/OAuthLoginPage.tsx

```diff
@@ -119,27 +282,76 @@ export const OAuthLoginPage = () => {
     }
     return null
   }, [redirect_uri])
-  const account = useSelector(getAccountUser)
-  const isLoggedIn = Boolean(account)
-  const [userEmail, setUserEmail] = useState<string | null>(null)
-  useEffect(() => {
-    const getAndSetEmail = async () => {
-      let email: string
+
+  const isRedirectValid = useMemo(() => {
+    return getIsRedirectValid({ parsedRedirectUri, redirectUri: redirect_uri })
```

Comment:
just double checking this is just to avoid computation rather than avoiding rerenders?

Reply:
yeah!